include/exadg/fluid_structure_interaction_precice/dof_interface.h

```diff
@@ -0,0 +1,200 @@
+#pragma once
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <exadg/fluid_structure_interaction_precice/coupling_interface.h>
+#include <exadg/fluid_structure_interaction_precice/dof_tools_extension.h>
+
+namespace Adapter
+{
+using namespace dealii;
+
+/**
+ * Derived class of the CouplingInterface: the classical coupling approach,
+ * where each participant defines an interface based on the locally owned
+ * triangulation. Here, dof support points are used for reading and writing.
+ * data_dim is equivalent to n_components, indicating the type of your data in
+ * the preCICE sense (Vector vs Scalar)
+ */
+template<int dim, int data_dim, typename VectorizedArrayType>
+class DoFInterface : public CouplingInterface<dim, data_dim, VectorizedArrayType>
+{
+public:
+  DoFInterface(std::shared_ptr<const MatrixFree<dim, double, VectorizedArrayType>> data,
+               std::shared_ptr<precice::SolverInterface>                           precice,
+               std::string                                                         mesh_name,
+               types::boundary_id                                                  interface_id,
+               int                                                                 mf_dof_index)
+    : CouplingInterface<dim, data_dim, VectorizedArrayType>(data, precice, mesh_name, interface_id),
+      mf_dof_index(mf_dof_index)
+  {
+  }
+
+  /**
+   * @brief define_mesh_vertices Define a vertex coupling mesh for preCICE
+   *        coupling the classical preCICE way
+   */
+  virtual void
+  define_coupling_mesh(const std::vector<Point<dim>> & vec) override;
+
+  /**
+   * @brief write_data Evaluates the given @param data at the
+   *        quadrature_points of the defined mesh and passes
+   *        them to preCICE
+   *
+   * @param[in] data_vector The data to be passed to preCICE (absolute
+   *            displacement for FSI)
+   */
+  virtual void
+  write_data(const LinearAlgebra::distributed::Vector<double> & data_vector,
+             const std::string &                                data_name) override;
+
+
+private:
+  /// The preCICE IDs
+  std::vector<int> interface_nodes_ids;
+  /// The deal.II associated IDs
+  std::vector<std::array<types::global_dof_index, data_dim>> global_indices;
+
+  bool interface_is_defined = false;
```

Comment:
Can't we remove this variable? `interface_node_ids.size() > 0` appears to be equivalent to `interface_is_defined == true`?

The same for `QuadInterface`.